### ISSUE TITLE
A goroutine memory leak

### ISSUE THREAD

**Issue by xuxiaofan1203** [godbus/dbus#383]
blocking position:
https://github.com/godbus/dbus/blob/76236955d466b078d82dcb16b7cf1dcf40ac25df/conn.go#L302
The reason is that after creating the ctx at
https://github.com/godbus/dbus/blob/76236955d466b078d82dcb16b7cf1dcf40ac25df/conn.go#L285
The goroutine has blocked at <-conn.ctx.Done(), so the conn.Close() can't execute to awaken it.
https://github.com/godbus/dbus/blob/76236955d466b078d82dcb16b7cf1dcf40ac25df/conn.go#L301-L304
you could reproduce the bug using goleak in the func TestSessionBus
https://github.com/godbus/dbus/blob/76236955d466b078d82dcb16b7cf1dcf40ac25df/conn_test.go#L15
like this
![6f82919286a79ef9691b9fbd1efada8](https://github.com/godbus/dbus/assets/61485003/363850f9-678f-48a7-b7a6-a69b49be41a7)
![a82c656440b461363e6c0f637d69b18](https://github.com/godbus/dbus/assets/61485003/5544a4fc-dfb0-4f3c-94e5-3a91b299bdca)

**Comment by Lzyyyds6** [comment]
Yes, I found the bug too. The newConn should be closed to awaken the ctx.Done().
